lib/dash/segment_template.js

```diff
@@ -94,61 +99,63 @@ shaka.dash.SegmentTemplate = class {
         }
       }
 
-      const references = SegmentTemplate.createFromTimeline_(
-          shallowCopyOfContext, info, initSegmentReference);
-
       const periodStart = context.periodInfo.start;
-      const periodEnd = context.periodInfo.duration ?
-          context.periodInfo.start + context.periodInfo.duration : Infinity;
-
-      /* When to fit segments.  All refactors should honor/update this table:
-       *
-       * | dynamic | infinite | last   | should | notes                     |
-       * |         | period   | period | fit    |                           |
-       * | ------- | -------- | ------ | ------ | ------------------------- |
-       * |     F   |     F    |    X   |    T   | typical VOD               |
-       * |     F   |     T    |    X   |    X   | impossible: infinite VOD  |
-       * |     T   |     F    |    F   |    T   | typical live, old period  |
-       * |     T   |     F    |    T   |    F   | typical IPR               |
-       * |     T   |     T    |    F   |    X   | impossible: old, infinite |
-       * |     T   |     T    |    T   |    F   | typical live, new period  |
-       */
-
-      // We never fit the final period of dynamic content, which could be
-      // infinite live (with no limit to fit to) or IPR (which would expand the
-      // most recent segment to the end of the presentation).
-      const shouldFit = !(context.dynamic && context.periodInfo.isLastPeriod);
```

Comment:
This logic is deleted, why?

Reply:
@theodab can you add you opinion here? Thanks!

Reply:
I can try modify the code a bit to be similar in structure to the regular segment index -- in other words I can include a more concrete/separate 'fit' step. That would make the table more applicable again.